python/upt-macports/Portfile

```diff
@@ -0,0 +1,36 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+PortSystem          1.0
+PortGroup           python 1.0
+
+name                upt-macports
+version             0.2.1
+revision            0
+
+maintainers         nomaintainer
+description         MacPorts backend for upt.
+long_description    ${description}
+
+platforms           darwin
+supported_archs     noarch
+
+homepage            https://github.com/mojca/upt-macports
+master_sites        https://test-files.pythonhosted.org/packages/83/89/cf55a2b5568d91375145db0c4ead0bd3b0bee81a76f87782c0b4ca26c220/
+distname            ${python.rootname}-${version}
+
+license             3-clause BSD License
+checksums           sha256  75f7ae203de1239057af647923d54c97b282f588251de0d845fe138b594ae7d2 \
+                    rmd160  bc1b90ed9c24417f4c37e0740efcf2cb5eec9f06 \
+                    size    4036
+                                
+python.versions     37
+
+
+depends_build-append \
+                    port:py${python.version}-setuptools
+
+depends_lib-append \
+                    port:upt \
```

Comment:
I didn't realize previously that frontends and backends need to depend on upt. Now that I think of it again, it kind of makes sense, but now I see a different "issue". Any user would would want to install `upt` would need to manually install each and every frontend (and backend) separately.

This makes me think of an alternative way. We could probably use `py-upt` as the main python module that other modules depend upon. But then we would make another port `upt` with a runtime dependency on all frontends and backends, and, like I suggested last time, the `upt` port could make a symlink from `upt` to `upt-py37`. What you call `upt` now would be called `py-upt`.

These are just some suggestions. I would be glad to hear opinion of others.

Reply:
I agree that all ```upt-*``` ports should be renamed to ```py-upt-*```, but see no reason to make another ```upt``` port. I would think that you either just add all the frontends and MacPorts backend unconditionally to the ```py-upt``` port as without frontends/backends it is not very useful. If you want to give people the option to install certain frontends you could add them as variants in the "main" ```py-upt``` port and, for example, by default install all of them.